src/basic/locale-util.c

```diff
@@ -260,7 +260,10 @@ bool locale_is_valid(const char *name) {
         if (!filename_is_valid(name))
                 return false;
 
-        if (!string_is_safe(name))
+        /* Locales look like: ll_CC.ENC@variant, where ll and
+         * CC are alphabetic, ENC is alphanumeric with dashes,
+         * and variant seems to be alphabetic. */
+        if (!in_charset(name, ALPHANUMERICAL "_.-@"))
```

Comment:
do you have a reference for this?

(i am fine with the change. I am just wondering. and would be great to have a link to the reference in the comment)